lib/index.js

```diff
@@ -133,62 +138,47 @@ export class Kadira {
   // are valid and correct. Data sent inside http headers.
   _checkAuth() {
     const uri = this._options.endpoint + '/ping';
-    const params = {headers: this._headers};
-    return this._send(uri, params);
+
+    const params = { headers: this._headers };
+
+    return axiosRetry(uri, params).then(res => {
+      const allowedContentTypes = res.headers['content-type'];
+
+      if (typeof allowedContentTypes === 'string' &&
+        allowedContentTypes.includes(ContentType.JSON_SEQ)) {
+        this._supportsJsonSeq = true;
+      }
+
+      return res.data;
+    });
   }
 
   // communicates with the server with http
   // Also handles response http status codes and retries
   _send(url, params) {
-    let retryEnabled = true;
+    return axiosRetry(url, {
+      ...params,
+      headers: {
+        ...this._headers,
+        ...params.headers,
+      },
+    }).then(res => res.data);
+  }
 
-    if (params.noRetry) {
-      retryEnabled = false;
-      delete params.noRetry;
+  _sendJsonStream(url, params) {
+    if (hasCircularReference(params.data)) {
```

Comment:
Why is this step necessary? How does zipson-stream handle circular references?

Reply:
It gives a maximum call stack reached error, but is not interceptable by the catch it seems.